mantle/kola/tests/ignition/qemufailure.go

```diff
@@ -45,47 +64,72 @@ func runIgnitionFailure(c cluster.TestCluster) {
 	}
 }
 
-func ignitionFailure(c cluster.TestCluster) error {
-	// We can't create files in / due to the immutable bit OSTree creates, so
-	// this is a convenient way to test Ignition failure.
-	failConfig, err := conf.EmptyIgnition().Render(conf.FailWarnings)
-	if err != nil {
-		return errors.Wrapf(err, "creating empty config")
+func runDualBootfsFailure(c cluster.TestCluster) {
+	if err := dualBootfsFailure(c); err != nil {
+		c.Fatal(err.Error())
 	}
-	failConfig.AddFile("/notwritable.txt", "Hello world", 0644)
+}
 
-	builder := platform.NewQemuBuilder()
-	defer builder.Close()
-	builder.SetConfig(failConfig)
-	err = builder.AddBootDisk(&platform.Disk{
-		BackingFile: kola.QEMUOptions.DiskImage,
-	})
+func runDualBootfsIgnitionFailure(c cluster.TestCluster) {
+	if err := dualBootfsIgnitionFailure(c); err != nil {
+		c.Fatal(err.Error())
+	}
+}
+
+// Read file and verify if it contains a pattern
+// 1. Read file, make sure it exists
+// 2. regex for pattern
+func fileHasPattern(tempLogFilePath string, searchPattern string) (bool, error) {
+	file, err := os.ReadFile(tempLogFilePath)
 	if err != nil {
-		return err
+		return false, err
 	}
-	builder.MemoryMiB = 1024
-	builder.Firmware = kola.QEMUOptions.Firmware
+	// File exists but it's empty
+	if len(file) == 0 {
+		return false, nil
+	}
```

Comment:
Minor/optional: do we even need to check for the empty case? I think the code below should already do the right thing in that case.

Reply:
Not really. I used it when problems occurred during development - will remove it here.